demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -53,3 +73,88 @@ std::string CTCGreedyDecoder(const std::vector<float> &data, const std::string&
     }
     return res;
 }
+
+std::string CTCBeamSearchDecoder(const std::vector<float> &data, const std::string& alphabet, char pad_symbol, double *conf, int bandwidth) {
+    std::string res = "";
+
+    const int num_classes = alphabet.length();
+
+    std::vector<BeamElement> curr;
+    std::vector<BeamElement> last;
+
+    last.push_back(BeamElement{std::vector<int>(), 1.f, 0.f});
+
+    for (std::vector<float>::const_iterator it = data.begin(); it != data.end(); it += num_classes) {
+        curr.clear();
+
+        std::vector<float> prob = std::vector<float>(num_classes, 0.f);
+        softmax(it, it + num_classes, prob);
+
+        for(auto& candidate: last) {
+            float prob_not_blank = 0.f;
+            std::vector<int> candidate_sentence = candidate.sentence;
```

Comment:
Could you use `const std::vector<int>&` here? It will allow to avoid copying of data (not so important here, just a bit better code style). Also it will emphasize for readers that this variable won't be changed (and point it to compiler -- just to avoid dummy bugs during future refactoring) 
```suggestion
            const std::vector<int>& candidate_sentence = candidate.sentence;
```